src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -324,7 +324,11 @@ private CompletionContext RewriteContext(CompletionContext context, RazorLanguag
             var invokeKind = (context as VSCompletionContext)?.InvokeKind;
             if (invokeKind.HasValue)
             {
-                rewrittenContext.InvokeKind = invokeKind.Value;
+                // The C# language server will not return any completions for the '@' character unless we
+                // send the completion request explicitly.
+                rewrittenContext.InvokeKind = languageKind == RazorLanguageKind.CSharp && context.TriggerCharacter == "@"
+                    ? VSCompletionInvokeKind.Explicit
```

Comment:
Shouldn't this be "typing"? Also, what's invokeKind evaluate to in this situation otherwise?

Reply:
By default it's Typing, so I believe we need to change it to Explicit. Roslyn won't return anything in the Typing case for '@'.

Reply:
I think this is because Roslyn will never bring up the completion list when '@' is typed unless explicitly invoked.
For example, given the code in a C# file:
```
var @switch = 1;
```
Typing '@' later on won't bring up a completion list until 's' is typed, even though we have a valid identifier starting with '@'.